crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -192,14 +193,51 @@ where
             Some(from) => from,
             None => return Err(SignError::NoAccount.into()),
         };
+
+        let chain_id = self.chain_id();
+        let nonce = self.get_transaction_count(from, None)?;
+        // we might need an oracle to get the gas limit of the current n/w as well if i m right?
+        let gas_limit = request.gas.unwrap_or_default();
+        // we need an oracle to fetch the gas price of the current chain
+        let gas_price = request.gas_price.unwrap_or_default();
+        let max_fee_per_gas = request.max_fee_per_gas.unwrap_or_default();
+
         let transaction = match request.into_typed_request() {
-            Some(tx) => tx,
+            Some(TypedTransactionRequest::Legacy(mut m)) => {
+                // nonce was not specified, we now need to update it to the latest nonce
+                if m.nonce.eq(&U256::ZERO) {
+                    m.nonce = nonce
+                }
```

Comment:
this we need to set beforehand, before building

just opened which is relevant here  https://github.com/paradigmxyz/reth/issues/1978

Reply:
you mean before building the tx type right?

Reply:
yes because this setting is universal